src/Shell/AclShell.php

```diff
@@ -155,7 +155,7 @@ public function delete()
 
         $identifier = $this->parseIdentifier($this->args[1]);
         $nodeId = $this->_getNodeId($class, $identifier);
-        $entity = $this->Acl->{$class}->newEntity(['id' => $nodeId]);
+        $entity = $this->Acl->{$class}->get($nodeId);
```

Comment:
This is a relatively expensive change (Since a DB query now needs to be made). Better would be to simply mark the entity as not new `newEntity(['id' => $nodeId], ['markNew' => false])`